### DISCUSSION TITLE
[SF][A11Y][ViewSettingsDialog] Several A11Y issues for ViewSettingsDialog

### DISCUSSION THREAD

**Discussion by ChristyFei** [UI5/webcomponents#10191] [Q&A]
Hi Colleagues,

The following issues have been reported by the A11Y team and cannot be fixed from the application side. Could you please review them and provide suggestions on how to address these issues?

1. Filter dialog title of View Settings has heading level 2 while dialog titles should start with heading level 1. 
<img width="1214" alt="Screenshot 2024-11-15 at 16 31 44" src="https://github.com/user-attachments/assets/68d45807-e900-4a42-bb05-23e2ae290ebf">

2. When filters are selected, the number of filters selected is displayed. However when focused, JAWS just announces, for example, "status 1". A screen reader user will not have context on what "1" means. The announcement must be "Status 1 filter selected" or something along those lines.

3. When Reset button is activated, the filters are reset but no indication to screen reader users of the same. Invisible message "filters reset" must be announced.

4. When select/unselect a filter item, Invisible message "Pending unselected" must be announced.

5. When keyboard navigation is moved to filter item, screen reader user doesn't know these fitler items are interactive. Provide an screen reader instruction to press ENTER to activate list items. 

Thank you,
Christy

**Comment by nnaydenow** [comment]
Hi @unazko,

Could you please check this discussion as acc expert responsible for the component?

**Reply by ChristyFei** [reply]
Is there any update for those issue?

**Comment by unazko** [comment]
Hello Christy,

Thanks for sharing your findings,
Here is what I've been able to analyze about the described issues:

1.  We'll need to discuss this with the accessibility experts. I see that we use `H1` in the `ui5-dialog` and in the sap.m.ViewSettingsDialog
from the openui5 project. If we change the heading level to `H1` however there will be an Access Assistant validation tool error
about an inappropriate jump in heading levels. In the meantime I've prepared the following [pull request](https://github.com/SAP/ui5-webcomponents/pull/10289) in order to sync up the heading levels with the `ui5-dialog` component.
2. This is a bug about missing group header, type and active state announcements. I've logged an [incident](https://github.com/SAP/ui5-webcomponents/issues/10297). We have the following text labelling each filter item in the openui5 project:
"**List Item . Is Active . Filter By . Field1**". Those will represent the list item type, indication that the item is active, a group header announcement and the list item's text.
3.  This should go through the design and accessibility colleagues so it's an enhancement request. The recommendation is logical. Couldn't find information in the specification for such announcement for the reset button.
4. There is a selected state change announcement already and the selected state is announced when user reaches the filter item.
This behavior is by design.
5. This is a bug about missing announcement covered in the note about issue 2.

Best regards,
Boyan

**Comment by ChristyFei** [comment]
Hi Boyan,

Thank you for answering my questions.

For points #2 and #5, could you please share the target version that will deliver this bug fix?
Regarding point #3, could you please share the discussion results with the accessibility colleagues?
For point #4, the unselected message isn't announced. Both selected and unselected messages should be announced.

Thank you,
Christy

**Comment by nnaydenow** [comment]
Hi @ChristyFei,

Bug fix mentioned in #1 has been released with version 2.5.0 which was released was week.

Update: by mistake #2 and #5 were marked as resolved

**Comment by unazko** [comment]
Hi @ChristyFei,

We've made the following [fix](https://github.com/SAP/ui5-webcomponents/pull/10320) in relation to point 3,
which will be with the next release (webc 2.7.0).

Best regards,
Boyan
